### PR TITLE
2.18.6

### PR DESCRIPTION
**Pull request by blublup**
Updates for dlt-daemon 2.18.6 compatibility.

### CODE REVIEW

**Comment by yen3** [comment]
@blublup HI, thanks for your pr.

Could you add the commit https://github.com/yen3/python-dlt/commit/05529817d2b1616a487c42491ee44e9e1c73106b ?
With the patch, we could test python-dlt for the current version and 2.1.8.6.
Then we could discuss how to make all tests pass.

Thank you.

File: dlt/dlt.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
     DLT_TYPE_INFO_UINT, DLT_TYPE_INFO_SINT, DLT_TYPE_INFO_STRG, DLT_TYPE_INFO_SCOD,
     DLT_TYPE_INFO_TYLE, DLT_TYPE_INFO_VARI, DLT_TYPE_INFO_RAWD,
     DLT_SCOD_ASCII, DLT_SCOD_UTF8, DLT_TYLE_8BIT, DLT_TYLE_16BIT, DLT_TYLE_32BIT, DLT_TYLE_64BIT,
-    DLT_TYLE_128BIT, DLT_DAEMON_TCP_PORT, DLT_CLIENT_RCVBUFSIZE,
+    DLT_TYLE_128BIT, DLT_DAEMON_TCP_PORT, DLT_RECEIVE_BUFSIZE,
```
Comment (yen3) [inline dlt/dlt.py:25]: I can not find the variable `DLT_RECEIVE_BUFSIZE ` on the original repo or your patch

File: dlt/dlt.py [hunk 2]
```diff
@@ -1072,7 +1080,7 @@ def py_dlt_client_main_loop(client, limit=None, verbose=0, dumpfile=None, callba
         # the status of the callback (in the case of dlt_broker, this is
         # the stop_flag Event), this loop will only proceed after the
         # function has returned or terminate when an exception is raised
-        recv_size = dltlib.dlt_receiver_receive(ctypes.byref(client.receiver), DLT_RECEIVE_SOCKET)
```
Comment (yen3) [inline dlt/dlt.py:1075]: Maybe we could keep this line

We could add the code block in the beginning of the file

```
from dlt.core import API_VER as API_VER_STR
API_VER = tuple(int(num) for num in API_VER_STR.split('.'))
```

then

```
if API_VER < (2, 18, 6):
    recv_size = dltlib.dlt_receiver_receive(ctypes.byref(client.receiver), DLT_RECEIVE_SOCKET)
else:
    recv_size = dltlib.dlt_receiver_receive(ctypes.byref(client.receiver))
```

File: dlt/dlt.py [hunk 3]
```diff
@@ -948,7 +963,10 @@ def connect(self, timeout=None):
                         # dlt_client_connect() behavior
                         connected = dltlib.dlt_receiver_init(ctypes.byref(self.receiver),
                                                              self.sock,
-                                                             DLT_CLIENT_RCVBUFSIZE)
+                                                             receiver_type,
+                                                             DLT_RECEIVE_BUFSIZE)
```
Comment (yen3) [inline dlt/dlt.py:967]: Does the function interface change between 2.18.5 and 2.18.6?
If yes, please call it by version. (The example is as the below.)

**Comment by yen3** [comment]
I try to mention some problems. After these problems are fixed, we could discuss how to process the PR.

Thanks for your contribution. :)

**Comment by yen3** [comment]
@blublup hi, do you want to continue to fix the patch?

**Comment by LocutusOfBorg** [comment]
Hello @yen3 I think I took care of the changes in
https://github.com/bmwcarit/python-dlt/pull/34

**Comment by yen3** [comment]
@LocutusOfBorg Thanks for your help and contribution. I will review it.

**Pull request closed without merging**